projects/plugins/social/src/class-note.php

```diff
@@ -196,8 +196,32 @@ public function maybe_flush_rewrite_rules( $force = false ) {
 	 * @param array $post_id The Post ID.
 	 */
 	public function override_empty_title( $title, $post_id ) {
-		if ( get_post_type( $post_id ) === self::JETPACK_SOCIAL_NOTE_CPT ) {
-			return wp_trim_words( get_the_excerpt(), 10 );
+		$post = get_post( $post_id );
+
+		if ( isset( $post ) && get_post_type( $post_id ) === self::JETPACK_SOCIAL_NOTE_CPT ) {
+			$post_date = get_the_date( '', $post );
+			$post_time = get_the_time( '', $post );
+
+			$label = __( 'Daily note', 'jetpack-social' );
+
+			/**
+			 * Filters the default label for Social Notes.
+			 *
+			 * @param string   $label The default label.
+			 * @param \WP_Post $post  The current post object.
+			 */
+			$label = apply_filters( 'jetpack_social_notes_default_label', $label, $post );
+
+			$title = sprintf( '%s %s %s', $label, $post_date, $post_time );
+
+			/**
+			 * Filters the title for Social Notes.
+			 *
+			 * @param string   $title The generated title (e.g. 'Daily note 29 Mar 2025 03:32 pm').
+			 * @param \WP_Post $post  The current post object.
+			 * @param string   $label The default label for the note.
+			 */
+			return apply_filters( 'jetpack_social_notes_default_title', $title, $post, $label );
```

Comment:
Since we have the full `$post`, I believe you should be able to rely on that instead of calling `get_the_date` and `get_the_time`. I would also recommend ensuring that the date will be properly displayed in all languages.

If we do all of this in one variable, we should be able to offer a single filter that would allow folks to customize the default title.

Something like this:

```suggestion
		if (
			$post instanceof \WP_Post &&
			self::JETPACK_SOCIAL_NOTE_CPT === $post->post_type
		) {
			$publishing_date = new \DateTimeImmutable(
				$post->post_date,
				wp_timezone()
			);

			$title = sprintf(
				/* Translators: placeholder is a fully-formatted date. */
				__( 'Social note, %1$s', 'jetpack-social' ),
				wp_date(
					get_option( 'date_format' ) . ' \a\t ' . get_option( 'time_format' ),
					$publishing_date->getTimestamp()
				)
			);

			/**
			 * Filters the default title for a Social Note.
			 *
			 * @since $$next-version$$
			 *
			 * @param string $title The default title.
			 * @param \WP_Post $post The post.
			 */
			$title = apply_filters( 'jetpack_social_notes_default_title', $title, $post );
```